distributed/tests/test_scheduler.py

```diff
@@ -313,10 +313,15 @@ def func(scheduler):
     await comm.close()
 
 
-def test_scheduler_init_pulls_blocked_handlers_from_config():
+@pytest.mark.asyncio
+async def test_scheduler_init_pulls_blocked_handlers_from_config():
+    # This test is async to allow us to properly close the scheduler since the
+    # init is already opening sockets for the HTTP server See also
+    # https://github.com/dask/distributed/issues/4806
     with dask.config.set({"distributed.scheduler.blocked-handlers": ["test-handler"]}):
         s = Scheduler()
     assert s.blocked_handlers == ["test-handler"]
+    await s.close()
```

Comment:
```suggestion
    with dask.config.set({"distributed.scheduler.blocked-handlers": ["test-handler"]}):
        async with Scheduler() as s:
            assert s.blocked_handlers == ["test-handler"]
```

Reply:
Based on the comment earlier in this test, it looks like we want to check that `Scheduler.__init__` handles blocking handlers. Using `async with Scheduler() as s:` will also call `start` which makes it difficult to isolate `Scheduler.__init__` behavior

Reply:
OK